pkg/scd/operational_intents_handler.go

```diff
@@ -396,11 +400,42 @@ func validateAndReturnUpsertParams(
 		return nil, stacktrace.NewError("Missing required UssBaseUrl")
 	}
 
+	valid.ussBaseURL = string(params.UssBaseUrl)
+
+	if params.SubscriptionId != nil {
+		valid.subscriptionID, err = dssmodels.IDFromOptionalString(string(*params.SubscriptionId))
+		if err != nil {
+			return nil, stacktrace.NewError("Invalid ID format for Subscription ID: `%s`", *params.SubscriptionId)
+		}
+	}
+
+	if params.NewSubscription != nil {
+		// The spec states that NewSubscription.UssBaseUrl is required and an empty value
+		// makes no sense, so we will fail if an implicit subscription is requested but the base url is empty
+		if params.NewSubscription.UssBaseUrl == "" {
+			return nil, stacktrace.NewError("Missing required USS base url for new subscription (in parameters for implicit subscription)")
+		}
+		if params.SubscriptionId != nil {
```

Comment:
That is checked below (although only a subscription is required). Adding this check here cover more cases but is also duplicate with below. If this behavior change is desired it should be in a separate PR. But I'm not sure that's really necessary, so I'd propose to just remove this check here.

Reply:
> That is checked below (although only a subscription is required)

Did you mean that this is checked above? (It used to be checked below but I figured I'd move the format check before this check)

I figured that I'd rather fail early here where we are actually validating the parameters: this might indeed be changing the behavior of the DSS and I can split that to a separate PR for clarity.

Reply:
Done ine https://github.com/interuss/dss/pull/1097

Reply:
Thanks for #1097, but was not removed from this one?

Reply:
I was expecting to get #1097 merged first, then keep the reordering of the checks in here (and have not yet rebased)